db/indexes.go

```diff
@@ -293,6 +298,15 @@ func (i *SGIndex) createIfNeeded(bucket base.N1QLStore, useXattrs bool, numRepli
 		return false, nil
 	}
 
+	// only create allDocs Index if EnableStarChannel is set to true
+	if i.simpleName == sgIndexes[IndexAllDocs].simpleName {
+		if !EnableStarChannelLog {
```

Comment:
I see the benefit of doing this check here (can set IsAllDocsIndexCreated), but calling createIfNeeded when we don't actually want to create the index feels counter-intuitive.  I think it would be preferable to have separate functionality to perform the check if the index exists for indexes we don't need to create.